library/bignum.c

```diff
@@ -1137,6 +1137,7 @@ void mpi_mul_hlp( size_t i, mbedtls_mpi_uint *s, mbedtls_mpi_uint *d, mbedtls_mp
         MULADDC_STOP
     }
 #else /* MULADDC_HUIT */
+#if !defined(MBEDTLS_SIZE_OPTIMIZED_MULADDC)
     for( ; i >= 16; i -= 16 )
```

Comment:
Does the manual loop unrolling here actually improve performance? Compilers can unroll loop on their own. A 4kB code block is non-negligible pressure on the instruction cache, so forced unrolling may hurt performance. We should benchmark this on different processors. Maybe we should just remove this block altogether!